lib/sqlalchemy/ext/asyncio/session.py

```diff
@@ -1436,6 +1436,40 @@ async def main():
 
     class_: Type[_AS]
 
+    @overload
+    def __init__(
+        self: "async_sessionmaker[AsyncSession]",
+        bind: Optional[_AsyncSessionBind] = ...,
+        *,
+        autoflush: bool = ...,
+        expire_on_commit: bool = ...,
+        info: Optional[_InfoType] = ...,
+        **kw: Any,
+    ):
+        ...
+    @overload
+    def __init__(
+        self,
+        bind: Optional[_AsyncSessionBind],
+        class_: Type[_AS],  # type: ignore
```

Comment:
There's a bit of awkwardness here, with the bind parameter being allowed as positional and keyword. So, there's this version without the default value, and the below version with the default value. Not sure if that can be merged into 1 overload somehow...